packages/adblocker/src/filters/network.ts

```diff
@@ -725,14 +725,23 @@ export default class NetworkFilter implements IFilter {
       // --------------------------------------------------------------------- //
       // parseOptions
       // --------------------------------------------------------------------- //
+      const denyallowEntities: Set<string> = new Set();
       for (const rawOption of getFilterOptions(line, optionsIndex + 1, line.length)) {
         const negation = rawOption[0].charCodeAt(0) === 126; /* '~' */
         const option = negation === true ? rawOption[0].slice(1) : rawOption[0];
         const value = rawOption[1];
 
         switch (option) {
+          case 'to':
           case 'denyallow': {
-            denyallow = Domains.parse(value.split('|'), debug);
+            let parts = value.split('|');
+            if (option === 'to') {
+              parts = parts.map((part) =>
+                part.charCodeAt(0) === 126 /* '~' */ ? part.slice(1) : `~${part}`,
```

Comment:
put that is a single loop with forEach below

Reply:
fixed in b9004e50ae696803b3a39ef4c42f660caf6560cc